### PR TITLE
Camera fly to with heading, pitch, roll

### PR DESCRIPTION
**Pull request by bagnell**
* Added `Camera.flyToHeadingPitchRoll`.
* Fixed the order of applying the heading, pitch and roll rotations. The roll and pitch angles would rotate around the same axis if heading is set to 0.0 or 180.0 degrees. Here is an example that you can paste into Sandcastle that works in this branch but not in master:
```javascript
var position = Cesium.Cartesian3.fromDegrees(-117.16, 32.71, 5000.0);
var heading =  Cesium.Math.toRadians(180.0);
var pitch = Cesium.Math.toRadians(-45.0);
var roll = Cesium.Math.toRadians(45.0);

viewer.camera.setView({
    position : position,
    heading : heading,
    pitch : pitch,
    roll : roll
});
```

### CODE REVIEW

**Comment by pjcozzi** [comment]
Can we add this to the camera tutorial?

**Comment by pjcozzi** [comment]
Are we sure this is the right design:
```
Camera.prototype.flyTo
Camera.prototype.flyToRectangle
Camera.prototype.flyToHeadingPitchRoll
```
Compared to
```
Camera.prototype.flyTo({
    destination : {
        point : {
            position : // ...
            direction : // ...
            up : // ...
        },
        rectangle : // ...
        needs-a-good-name : {
            position : // ...
            heading : // ...
            pitch : // ...
            roll : // ...
       }
    }
});

```
Have we thought about what is less code, tests, and doc, and is easier for users to learn and use?

I haven't looked closely, but want to make sure we thought this through.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another review.

I deprecated `flyToRectangle`. You can now do
```javascript
camera.flyTo({
    destination : Cartesian3.fromDegrees(/* ... */)
});
```
or
```javascript
camera.flyTo({
    destination : Rectangle.fromDegrees(/* ... */)
});
```
For the end orientation you can do
```javascript
camera.flyTo({
    destination : Cartesian3.fromDegrees(/* ... */),
    orientation : {
        direction : /* ... */,
        up : /* ... */
    }
});
```
or 
```javascript
camera.flyTo({
    destination : Cartesian3.fromDegrees(/* ... */),
    orientation : {
        heading : /* ... */,
        pitch : /* ... */,
        roll : /* ... */
    }
});
```

File: CHANGES.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ Change Log
   * `Camera.tilt` was deprecated in Cesium 1.6. It will be removed in Cesium 1.7. Use `Camera.pitch`.
   * `Camera.heading` and `Camera.tilt` were deprecated in Cesium 1.6. They will become read-only in Cesium 1.7. Use `Camera.setView`.
   * `Camera.setPositionCartographic` was deprecated in Cesium 1.6. It will be removed in Cesium 1.7. Use `Camera.setView`.
+  * The `direction` and `up` options to `Camera.flyTo` have been deprecated in Cesium 1.6. They will be removed in Cesium 1.8. Use the `orientation` option.
```
Comment (pjcozzi) [inline CHANGES.md]: Submit an issue for these.

**Comment by pjcozzi** [comment]
Looks good.

**Pull request merged**